modules/omsBidAdapter.js

```diff
@@ -0,0 +1,321 @@
+      }
+
+      return imp;
+    })
+
+    const referrer = bidderRequest?.refererInfo?.page || '';
+    const publisherId = getBidIdParameter('publisherId', bidReqs[0].params);
+
+    const payload = {
+      id: getUniqueIdentifierStr(),
+      imp: impressions,
+      site: {
+        domain: bidderRequest?.refererInfo?.domain || '',
+        page: referrer,
+        publisher: {
+          id: publisherId
+        }
+      },
+      device: {
+        devicetype: _getDeviceType(),
+        w: screen.width,
+        h: screen.height
+      },
+      tmax: bidderRequest?.timeout
+    };
+
+    if (bidderRequest?.gdprConsent) {
+      deepSetValue(payload, 'regs.ext.gdpr', +bidderRequest.gdprConsent.gdprApplies);
+      deepSetValue(payload, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+    }
+
+    if (bidderRequest?.uspConsent) {
+      deepSetValue(payload, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+    }
+
+    if (config.getConfig('coppa') === true) {
+      deepSetValue(payload, 'regs.coppa', 1);
+    }
+
+    if (bidReqs?.[0]?.schain) {
+      deepSetValue(payload, 'source.ext.schain', bidReqs[0].schain)
+    }
+
+    if (bidReqs?.[0]?.userIdAsEids) {
+      deepSetValue(payload, 'user.ext.eids', bidReqs[0].userIdAsEids || [])
+    }
+
+    if (bidReqs?.[0].userId) {
+      deepSetValue(payload, 'user.ext.ids', bidReqs[0].userId || [])
+    }
+
+    return {
+      method: 'POST',
+      url: URL,
+      data: JSON.stringify(payload),
+    };
+  } catch (e) {
+    logError(e, {bidReqs, bidderRequest});
+  }
+}
+
+function isBidRequestValid(bid) {
+  if (bid.bidder !== BIDDER_CODE || !bid.params || !bid.params.publisherId) {
+    return false;
+  }
+
+  return true;
+}
+
+function interpretResponse(serverResponse) {
+  let response = [];
+  if (!serverResponse.body || typeof serverResponse.body != 'object') {
+    logWarn('OMS server returned empty/non-json response: ' + JSON.stringify(serverResponse.body));
+    return response;
+  }
+
+  const {body: {id, seatbid}} = serverResponse;
+
+  try {
+    if (id && seatbid && seatbid.length > 0 && seatbid[0].bid && seatbid[0].bid.length > 0) {
+      response = seatbid[0].bid.map(bid => {
+        return {
+          requestId: bid.impid,
+          cpm: parseFloat(bid.price),
+          width: parseInt(bid.w),
+          height: parseInt(bid.h),
+          creativeId: bid.crid || bid.id,
+          currency: 'USD',
+          netRevenue: true,
+          mediaType: BANNER,
+          ad: _getAdMarkup(bid),
+          ttl: 60,
+          meta: {
+            advertiserDomains: bid?.adomain || []
+          }
+        };
+      });
+    }
+  } catch (e) {
+    logError(e, {id, seatbid});
+  }
+
+  return response;
+}
+
+// Don't do user sync for now
+function getUserSyncs(syncOptions, responses, gdprConsent) {
+  return [];
+}
+
+function onTimeout(timeoutData) {
+  if (timeoutData === null) {
+    return;
+  }
+
+  _trackEvent('timeout', timeoutData);
```

Comment:
Please put this behind configuration, publishers don't always want to provide

Reply:
@patmmccann Can you provide please example how to put it behind the configuration?

Reply:
@patmmccann just a friendly reminder. Could you please provide an example as requested?

Reply:
just check the params for something like disableTimeoutReporting and if it is there return true

Reply:
We decided to remove this event tracking